support-lib/jni/Marshal.hpp

```diff
@@ -178,6 +178,25 @@ namespace djinni
         }
     };
 
+    struct WString
+    {
+        using CppType = std::wstring;
+        using JniType = jstring;
+
+        using Boxed = WString;
+
+        static CppType toCpp(JNIEnv* jniEnv, JniType j)
+        {
+            assert(j != nullptr);
+            return jniWStringFromString(jniEnv, j);
+        }
+
+        static LocalRef<JniType> fromCpp(JNIEnv* jniEnv, const CppType& c)
+        {
+            return {jniEnv, jniStringFromWString(jniEnv, c.c_str())};
```

Comment:
Good point, though it looks like the 8-bit version above has the same issue when it calls jniStringFromUTF8().  That seems like a bug worth fixing.